iele-binary.md

```diff
@@ -5,104 +5,302 @@ Here we define an ad-hoc binary encoding for IELE. This encoding is subject to c
+                   | SHA3 ()
+                   | MLOAD ()
+                   | MSTORE ()
+                   | SLOAD ()
+                   | EXTCODESIZE ()
+                   | BLOCKHASH ()
+                   | BALANCE ()
+                   | SSTORE ()
+                   | MOVE ()
+                   | LOG1 ()
+
+    syntax TernOp ::= ADD ()
+                    | MUL ()
+                    | SUB ()
+                    | DIV ()
+                    | MOD ()
+                    | EXP ()
+                    | SIGNEXTEND ()
+                    | TWOS ()
+                    | LT ()
+                    | LE ()
+                    | GT ()
+                    | GE ()
+                    | EQ ()
+                    | NE ()
+                    | AND ()
+                    | OR ()
+                    | XOR ()
+                    | BYTE ()
+                    | LOG2 ()
+
+    syntax QuadOp ::= ADDMOD ()
+                    | MULMOD ()
+                    | EXPMOD ()
+                    | MLOADN ()
+                    | MSTOREN ()
+                    | LOG3 ()
+
+    syntax FiveOp ::= LOG4 ()
+
+    syntax CreateOp ::= CREATE ( Int , Int )
+
+    syntax CopyCreateOp ::= COPYCREATE ( Int )
+
+    syntax CallOp ::= CALL ( Int , Int , Int )
+                    | CALLCODE ( Int , Int , Int )
+
+    syntax CallSixOp ::= DELEGATECALL ( Int , Int , Int )
+                       | STATICCALL ( Int , Int , Int )
+
+    syntax ReturnOp ::= RETURN ( Int )
+                      | REVERT ( Int )
+
+    syntax LocalCallOp ::= LOCALCALL ( Int , Int , Int )
+
+    syntax OpCode ::= NullOp
+                    | UnOp
+                    | CreateOp
+                    | CopyCreateOp
+                    | CallOp
+                    | CallSixOp
+                    | ReturnOp
+                    | LocalCallOp
+                    | TernOp
+                    | QuadOp
+                    | BinOp
+                    | FiveOp
+
+    syntax Contract ::= #dasmContract ( WordStack , IeleName )       [function]
+                      | #dasmContract ( WordStack , Int , Map, IeleName , TopLevelDefinitions, Int , Int ) [function, klabel(#dasmContractAux)]
+ // -----------------------------------------------------------------------------
+    rule #dasmContract( .WordStack, _) => #emptyCode
+    rule #dasmContract( 99 : NBITS : WS, NAME ) => #dasmContract(WS, NBITS, 0 |-> "init", NAME, .TopLevelDefinitions, 1, #sizeWordStack(WS) +Int 2)
+    rule #dasmContract( 105 : W1 : W2 : WS, NBITS, FUNCS, NAME, DEFS, N, SIZE ) => #dasmContract(#drop(W1 *Int 256 +Int W2, WS), NBITS, N |-> #parseToken("IeleName", #unparseByteStack(#take(W1 *Int 256 +Int W2, WS))) FUNCS, NAME, DEFS, N +Int 1, SIZE )
+    rule #dasmContract( 106 : W1 : W2 : WS, NBITS, FUNCS, NAME, DEFS, N, SIZE ) => #dasmContract(#take(W1 *Int 256 +Int W2, WS), NAME ++IeleName N) ++Contract #dasmContract(#drop(W1 *Int 256 +Int W2, WS), NBITS, FUNCS, NAME, external contract N DEFS, N +Int 1, SIZE)
```

Comment:
This gives the child contract the name `NAME ++IeleName N` but gives the parent contract a reference to name `N`. That won't work with the current implementation of the create call.